File: JavaScript_Courses/JavaScript_Basics/JS_BP_2022_10/06_Nested_Loops_Exercise/06_Cinema_Tickets.js
function cinemaTickets(input) {  // ["Taxi","10","standard","kid","student","student","standard","standard","End",
                                 //  "Scary Movie","6","student","student","student","student", "student","student",
                                 // "Finish"]
    let studentCurrent = 0;
    let stardardCurrent = 0;
    let kidCurrent = 0;

    let totalTickets = 0;


    let i = 0;
    let currentMovie = input[i];
    i++;

    while (currentMovie != 'Finish') {
        let seatAvailable = Number(input[i]);
        i++;
        let seatBougth = 0;

        while (true) {
            if (seatAvailable == seatBougth) {
                break;
            }

            let currentTicket = input[i];  // ("student", "standard", "kid")  "End"
            i++;

            if (currentTicket == "End") {
                break;
            }

            seatBougth++;
            totalTickets++;
            if (currentTicket == "student") {
                studentCurrent++;
            } else if (currentTicket == "standard") {
                stardardCurrent++;
            } else if (currentTicket == "kid") {
                kidCurrent++;
            }
        
        }


        console.log(`${currentMovie} - ${(100 * seatBougth / seatAvailable).toFixed(2)}% full.`);


        currentMovie = input[i];
        i++;
    }


    console.log(`Total tickets: ${totalTickets}`);  // studentCurrent + stardardCurrent + kidCurrent
    console.log(`${(100 * studentCurrent / totalTickets).toFixed(2)}% student tickets.`);
    console.log(`${(100 * stardardCurrent / totalTickets).toFixed(2)}% standard tickets.`);
    console.log(`${(100 * kidCurrent / totalTickets).toFixed(2)}% kids tickets.`);

}


// cinemaTickets(["Taxi",
//     "10",
//     "standard",
//     "kid",
//     "student",
//     "student",
//     "standard",
//     "standard",
//     "End",
//     "Scary Movie",
//     "6",
//     "student",
//     "student",
//     "student",
//     "student",
//     "student",
//     "student",
//     "Finish"]);

File: Python_Courses/Python_Basics/Python_PB_2023_10/06_Nested_Loops_Exercise/06_Cinema_Tickets.py
student_tickets = 0
standard_tickets = 0
kid_tickets = 0

command = input()  # "Finish" OR movie_name
while command != "Finish":
    movie_name = command
    total_seats = int(input())

    ticket_bought = 0
    ticket_type = input()  # "End" OR ticket_type IN ("student", "standard", "kid")
    while ticket_type != "End":
        ticket_bought += 1

        if ticket_type == "student":
            student_tickets += 1
        elif ticket_type == "standard":
            standard_tickets += 1
        elif ticket_type == "kid":
            kid_tickets += 1

        if ticket_bought == total_seats:
            break

        ticket_type = input()

    percent_full = ticket_bought / total_seats * 100
    print(f"{movie_name} - {percent_full :.2f}% full.")

    command = input()


total_tickets = student_tickets + standard_tickets + kid_tickets
print(f"Total tickets: {total_tickets}")
print(f"{student_tickets / total_tickets * 100 :.2f}% student tickets.")
print(f"{standard_tickets / total_tickets * 100 :.2f}% standard tickets.")
print(f"{kid_tickets / total_tickets * 100 :.2f}% kids tickets.")
